src/App/Pages/Vault/CipherAddEditPageViewModel.cs

```diff
@@ -605,6 +605,8 @@ public async Task GenerateUsernameAsync()
                 return;
             }
 
+            var website = (bool)(Cipher?.Login?.HasUris) ? Cipher?.Login?.Uris[0].Host : null;
```

Comment:
If `Cipher` or `Login` is `null` then when you cast it to `bool` it will throw an exception, so it should be something like:
```c#
var website = (Cipher?.Login?.HasUris ?? false) ? Cipher.Login.Uris[0].Host : null;
```
or another even better way:
```c#
var website = Cipher?.Login?.Uris?.FirstOrDefault()?.Host;
```

Reply:
Fixed with second suggestion, thank you for your help.